### ISSUE TITLE
Using Julia_jll gives a weird warning

### ISSUE THREAD

**Issue by fingolfin** [JuliaPackaging/BinaryBuilder.jl#854]
I finally want to build a JLL for our project which links against Julia, so I tried using the BinaryBuilder wizard to produce an initial build_tarbals.jl. I specified Julia_jll as a dependency, among others (e.g. GMP_jll). But now I see this and I don't understand what it means:
```
┌ Warning: Dependency Julia_jll does not have a mapping for Julia!
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/9oqch/src/Prefix.jl:450
```
Moreover, in the build environment, there is no `julia` executable nor `libjulia` nor the headers. Other JLLs like GMP_jll are present and work fine, though.

A bit longer excerpt of build log in case it helps (I am using Julia 1.4.2 and BinaryBuilder v0.2.5 in case it matters):
```
   Updating registry at `~/.julia/registries/General`
   Updating git-repo `https://github.com/JuliaRegistries/General.git`
  Resolving package versions...
Downloading artifact: GMP
######################################################################## 100.0%##O#-#                               Downloading artifact: Readline
######################################################################## 100.0%##O#-#                               Downloading artifact: Gettext
######################################################################## 100.0%##O#-#                               Downloading artifact: Zlib
######################################################################## 100.0%##O#-#                               Downloading artifact: Libiconv
######################################################################## 100.0%##O=#  #                             Downloading artifact: XML2
######################################################################## 100.0%##O#-#                               Downloading artifact: Ncurses
######################################################################## 100.0%##O#-#                                  Updating `/tmp/jl_ucy4il/rIm7KyTe/.project/Project.toml`
  [781609d7] + GMP_jll v6.1.2+5
  [244e23d5] + Julia_jll v1.4.2+0
  [05236dd9] + Readline_jll v8.0.4+0
   Updating `/tmp/jl_ucy4il/rIm7KyTe/.project/Manifest.toml`
  [781609d7] + GMP_jll v6.1.2+5
  [78b55507] + Gettext_jll v0.20.1+4
  [244e23d5] + Julia_jll v1.4.2+0
  [94ce4f54] + Libiconv_jll v1.16.0+5
  [68e3532b] + Ncurses_jll v6.1.0+4
  [05236dd9] + Readline_jll v8.0.4+0
  [02c8fc9c] + XML2_jll v2.9.10+1
  [83775a58] + Zlib_jll v1.2.11+14
  [2a0f44e3] + Base64
  [ade2ca70] + Dates
  [b77e0a4c] + InteractiveUtils
  [76f85450] + LibGit2
  [8f399da3] + Libdl
  [56ddb016] + Logging
  [d6f4376e] + Markdown
  [44cfe95a] + Pkg
  [de0858da] + Printf
  [3fa0cd96] + REPL
  [9a3f8284] + Random
  [ea8e919c] + SHA
  [9e88b42a] + Serialization
  [6462fe0b] + Sockets
  [cf7118a7] + UUIDs
  [4ec0a83e] + Unicode
┌ Warning: Dependency Julia_jll does not have a mapping for Julia!
└ @ BinaryBuilder ~/.julia/packages/BinaryBuilder/9oqch/src/Prefix.jl:450
```

**Comment by fingolfin** [comment]
Besides resolving the issue, perhaps also the message "Dependency Julia_jll does not have a mapping for Julia" could be improved. It's unclear to me whether "Julia" here is a filename? A variable name? Something else? What is missing? What kind of mapping is this talking about?!

**Comment by giordano** [comment]
Note this package for the time being isn't really meant to be "used", it should only be a build dependency of packages that need to build against libjulia

**Comment by fingolfin** [comment]
Yeah I want to use libjulia. But the warning concerns me; and I actually cannot find libjulia nor julia.h in here.

Btw nice webinar, thanks for that :-)

**Comment by giordano** [comment]
> Yeah I want to use libjulia.

From within Julia, you should probably use the libjulia that comes with your Julia instead.  Loading another (possibly different/incompatible) libjulia inside Julia is calling for a disaster :stuck_out_tongue:

**Comment by fingolfin** [comment]
Of course I want to link against the user's installed linjulia *during runtime* bit during build time, I still need a libjulia to make the linker happy. This was already discussed in https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/511. I know wanted to try to implement this, now that Julia_jll exists, but run into the above problem.

**Comment by giordano** [comment]
Try to require GCC v5

**Comment by fingolfin** [comment]
Yeah, inspecting the JLL revealed that it has specific requirements (libgfortran4 and cxx11), so I mimicked other packages using it. The result is https://github.com/JuliaPackaging/Yggdrasil/pull/1342

With that in mind, https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/14 tries to improve the error message over which I stumbled; feedback on that would be very much welcome :-).

Thank you very much for your support and your work on BinaryBuilder!
